packages/stream-node/writer.js

```diff
@@ -57,7 +57,7 @@ export const makeNodeWriter = writer => {
           } else {
             resolve(undefined);
           }
-        }),
+        }).then(() => ({ done: false, value: undefined })),
```

Comment:
We could harden and intern the medial iteration result.

We could also avoid some promise and closure allocations by changing `resolve(undefined)` to `resolve(nonFinalIteration)` and creating a `() => resolve(nonFinalIteration)` closure only if `writer.write` is false.

Reply:
Yeah I considered those as well, but went for this for simplicity's sake. If you have a preference I'm happy to go any other way.

Reply:
Fixed. Also hardened the `finalIteration` result as well.